packages/api/src/controllers/webhook.ts

```diff
@@ -188,6 +196,48 @@ app.put("/:id", authorizer({}), validatePost("webhook"), async (req, res) => {
   res.json({ id: req.body.id });
 });
 
+app.patch(
+  "/:id",
+  authorizer({}),
+  validatePost("webhook-patch-payload"),
+  async (req, res) => {
+    const webhook = await db.webhook.get(req.params.id);
+
+    if (!webhook) {
+      throw new NotFoundError(`webhook not found`);
+    }
+
+    if (
+      (webhook.userId !== req.user.id || webhook.deleted) &&
+      !req.user.admin
+    ) {
+      // do not reveal that webhooks exists
+      throw new NotFoundError(`webhook not found`);
+    }
+
+    const { id, userId, createdAt, kind } = webhook;
+
+    if (req.body.streamId) {
+      const stream = await db.stream.get(req.body.streamId);
+      if (!stream) {
+        throw new NotFoundError(`stream not found`);
+      }
+    }
+
+    await db.webhook.update(req.params.id, {
+      ...req.body,
+      // non editable fields
+      id,
+      userId,
+      createdAt,
+      kind,
+    });
```

Comment:
The `update` function works as a patch API itself, so you don't need to specify every field in the desired resulting object. So I believe a safer approach here, instead of "blocking" the uneditable fields, would be to explicitly say which fields are editable and setting only those instead. This will safeguard against potential bugs in the request payload schema validation (like for example forgetting an `additionalProperties: false`) and also be easier to evolve in the future as we add more fields. For example one field you forgot to list as uneditable here was the `deleted`, which is an internal field that should never be exposed to users.

So I think this would be safer instead:
```suggestion
    const { name, events, url, sharedSecret, streamId } = req.body;
    await db.webhook.update(req.params.id, {
      name, 
      events, 
      url, 
      sharedSecret, 
      streamId
    });
```
(undefined fields will also not be included in the request to the DB, so it won't clear their value if the user doesn't send them in the request!)

WDYT?